subprojects/core/src/integTest/groovy/org/gradle/normalization/ConfigureRuntimeClasspathNormalizationIntegrationTest.groovy

```diff
@@ -628,6 +668,23 @@ class ConfigureRuntimeClasspathNormalizationIntegrationTest extends AbstractInte
             return this
         }
 
+        PropertiesResource setComment(String comment) {
+            this.comment = comment
+            return this
+        }
+
+        PropertiesResource overwriteProperties(Map<String, String> propertiesMap) {
+            // Preserve the order of the properties in the map when writing the properties file
+            backingFile.withWriter {writer ->
+                writer.write("# ${comment}\n")
+                propertiesMap.each {key, value ->
+                    writer.write("${key}: ${value}\n")
```

Comment:
Doesn't this use the wrong format? `.properties` files should have `property = value` syntax, no?

We could also use `Properties.store()` here I think. It adds a comment with the current date, but we are testing normalization after all, so that shouldn't be a problem...

Reply:
heh...so this was totally unintentional (too much working with maps) but as it turns out, "key:value" is a valid (if rarely used) alternative format for properties files (see https://en.wikipedia.org/wiki/.properties).  I've changed it to "key=value" to avoid any confusion, but it was still testing the right thing.

As far as `Properties.store()` is concerned, I don't think there is any guarantee that order will be preserved when storing (which is important for the test that uses this method).  It's probably implementation specific, but at least in adoptopenjdk the properties implementation just uses a `HashMap` to store the entries.  There are other solutions here, but just writing the file seemed like a straightforward and simple solution.

Reply:
Wow. Been working with Java for decades now, and there are still surprises like that. :)